bims/static/js/views/olmap_layers.js

```diff
@@ -200,7 +203,8 @@ define(['shared', 'backbone', 'underscore', 'jquery', 'ol', 'views/layer_style']
                             params: {
                                 layers: value.wms_layer_name,
                                 format: value.wms_format
-                            }
+                            },
+                            crossOrigin: 'anonymous'
```

Comment:
@ann26 if you add this line then I couldn't access layers from maps.kartoza geoserver, here's the error message : 

![image](https://user-images.githubusercontent.com/1979569/48242693-fff93f80-e40e-11e8-8011-d8d5ca3c373c.png)


@NyakudyaA any solution for this ?

Reply:
@dimasciput let me see if I enable GeoServer CORS if it will work for these layers

Reply:
@dimasciput cc @ann26 can you try to access GeoServer layers now and see if you can access them

Reply:
@NyakudyaA I could access them, but if I show the layers from our GeoServer instances in testing and staging limpopobims, I will get the same error. What did you do on the GeoServer?  Maybe I can replicate the same process for our own instances.

Reply:
@ann26 If I choose Topography or Aerial basemap layer I will get this message : 
![image](https://user-images.githubusercontent.com/1979569/48456715-a029dc80-e7f2-11e8-98b7-829341657b8d.png)

And the basemap will not appear.

Reply:
hi @ann26 in my local it does not work from the beginning, this code didn't change the way they behave in my local. I always got cors error either way

Reply:
@dimasciput @NyakudyaA @gubuntu 
In my local, only biodiversity and Rivers NFEPA layers can be displayed (others I got 404) and the the basemap that is working is Topography, Openstreetmap, Aerial Photography and Bing (for terrain I got invalid key, for plain greyscale and B&W simply does not work I couldn't figure out why).

However due to cors error, the biodiversity layers can't be downloaded (although I remember it was working before) and the Topography and Aerial Photography basemap can't be downloaded either because of cors error (I tried to add `crossOrigin: 'anonymous'` but apparently it makes the map does not appear, so I removed the cross origin from them)

So the only layer and basemaps that I've tested and working to be downloaded are:
- Rivers NFEPA (layer)
- Openstreetmap (basemap)
- Bing (basemap)

Reply:
afaik LEDET overlays come only from the local GeoNode and hence are on the same domain so there should be no CORS. 

All the basemaps come from third parties, surely their CORS behaviour should be consistent and able to be overcome? Same if we fetch WMS or WFS from third parties like maps.kartoza.com. 

Also @ann26 what do you mean by 'downloaded' in the context of pdf generation?

Reply:
@gubuntu yes that's what I thought with the basemaps too, but apparently the ones that does not have cors problem are openstreetmap and bing.

I mean when we download the map, when the basemap or the layer has cors problem the result is an empty canvas as shown below:
![map 58](https://user-images.githubusercontent.com/26101337/48464810-62897b80-e813-11e8-9361-669e86d38276.png)

while the maps that is successfully downloaded should be like this:
(basemap: bing, layer: Rivers NFEPA)
![map 53](https://user-images.githubusercontent.com/26101337/48464831-7c2ac300-e813-11e8-92ec-8c91ac34c916.png)

Reply:
update: biodiversity layer has no cors problem